assets/js/base/components/block-error-boundary/index.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import BlockError from './block-error';
+import './style.scss';
+
+interface DerivedStateReturn {
+	errorMessage: JSX.Element | string;
+	hasError: boolean;
+}
+
+interface Error {
```

Comment:
Since `Error` is a default constructor type, I'd avoid overriding it. I'd call this a slightly different name perhaps?